### DISCUSSION TITLE
Is clicking the close box on a window supposed to generate a CloseRequested event.

### DISCUSSION THREAD

**Discussion by John-Nagle** [rust-windowing/winit#3841] [Q&A]
I don't seem to be getting **CloseRequested** events on Linux/X11 when the close "x" on the window is clicked. Or they're being consumed/lost in egui-winit, or something. 

Code looks like:

```
match event {
            winit::event::Event::WindowEvent {
                event: WindowEvent::Resized(size),
                ..
            } => {
                data.egui_routine
                    .resize(size.width, size.height, window.scale_factor() as f32);
                //  Tell camera about size change.
                data.render_camera_state_link
                    .lock() // brief lock for small scope
                    .unwrap()
                    .update_window_size((size.height as u16, size.width as u16));
            }
            winit::event::Event::WindowEvent {
                event: WindowEvent::CloseRequested {},
                ..
            } => {
                log::warn!("Quitting, window closed.");
                data.scene_quit.store(true, Ordering::Release); // tell other threads

    ...
```

I never get that "Quitting" log message. Other events work fine. Am I looking for the wrong event type?

**Comment by John-Nagle** [comment]
When in the window life cycle do these events appear?

- CloseRequested -- The window has been requested to close. 
- Destroyed -- The window has been destroyed.

When is it time to leave the event loop?

**Reply by kchibisov** [reply]
You should leave event loop when you want to do it, window being destroyed doesn't mean that you have to leave the loop, since some apps recreate the window later on, etc.

For general case you could do so once you _drop_ your window or once it got `destroyed`.

**Comment by John-Nagle** [comment]
I need to test some more, but I seem to be able to close the window without getting either **CloseRequested** or **Destroyed**. But **egui** may be eating those events; it gets first look at them and can consume them.

**Comment by kchibisov** [comment]
You can test with winit example, should be much simpler than egui.
